packages/core/http/core-http-server-internal/src/http_config.ts

```diff
@@ -290,7 +298,27 @@ export type HttpConfigType = TypeOf<typeof configSchema>;
 export const config: ServiceConfigDescriptor<HttpConfigType> = {
   path: 'server' as const,
   schema: configSchema,
-  deprecations: ({ rename }) => [rename('maxPayloadBytes', 'maxPayload', { level: 'warning' })],
+  deprecations: ({ rename }) => [
+    rename('maxPayloadBytes', 'maxPayload', { level: 'warning' }),
+    (settings, fromPath, addDeprecation, { docLinks }) => {
+      const cfg = get(settings, fromPath);
+      if (!cfg?.ssl?.enabled || cfg?.protocol === 'http1') {
+        addDeprecation({
+          level: 'warning',
+          title: `Consider enabling TLS and using HTTP/2 to improve security and performance.`,
+          configPath: `${fromPath}.protocol,${fromPath}.ssl.enabled`,
+          message: `TLS is not enabled, or the HTTP protocol is set to HTTP/1. Enabling TLS and using HTTP/2 improves security and performance.`,
+          correctiveActions: {
+            manualSteps: [
+              `Set up TLS by configuring ${fromPath}.ssl.`,
+              `Set the protocol to 'http2' by updating ${fromPath}.protocol to 'http2' in your configuration.`,
+            ],
+          },
+          documentationUrl: docLinks.settings + '#server-protocol',
```

Comment:
>          documentationUrl: docLinks.settings + '#server-protocol',

Rather than hacking the URL here, I think the new URL ought to be added to the doc link service, so that we can ensure it's updated appropriately when we migrate to the new docs system.